### PR TITLE
PI-2344 - add api

### PR DESCRIPTION
**Pull request by achimber-moj**

### CODE REVIEW

File: projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/controller/SubjectAccessRequestsController.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.service.SubjectAccessRequestsService
+
+@RestController
+@RequestMapping("probation-case/{crn}")
+@PreAuthorize("hasRole('PROBATION_API__SUBJECT_ACCESS_REQUEST___DETAIL')")
```
Comment (marcus-bcl) [inline projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/controller/SubjectAccessRequestsController.kt:12]: Minor thing - think you have an extra `_` here:
```suggestion
@PreAuthorize("hasRole('PROBATION_API__SUBJECT_ACCESS_REQUEST__DETAIL')")
```

File: projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integration/delius/person/entity/Person.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Column(name = "surname", length = 35)
+    val surname: String,
+
+    @Column(columnDefinition = "number")
+    val softDeleted: Boolean = false,
+) {
+    fun getName(): String {
+        return "$forename " + listOfNotNull(secondName, thirdName).joinToString(" ") + " $surname"
+    }
```
Comment (marcus-bcl) [inline projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integration/delius/person/entity/Person.kt:41]: If they have no middle names, you'll end up with a double space between forename and surname here.  Might be easier to include forename and surname in the `listOfNotNull` so all the parts are joined in the same way?
```suggestion
    fun getName() = listOfNotNull(forename, secondName, thirdName, surname).joinToString(" ")
```

File: projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/controller/SubjectAccessRequestsController.kt (hunk 1 above)
Comment (achimber-moj) [reply projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/controller/SubjectAccessRequestsController.kt:12]: done

File: projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integration/delius/person/entity/Person.kt (hunk 2 above)
Comment (achimber-moj) [reply projects/subject-access-requests-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integration/delius/person/entity/Person.kt:41]: done

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**